sdk/test/UnitTests/Custom/Runtime/RegionEndpointTests.cs

```diff
@@ -0,0 +1,118 @@
+    {
+        private static class SampleValues
+        {
+            public const string SampleRegionDisplayName = "US west too";
+            public const string SampleHostName = "host-name";
+            public const string SampleAuthRegion = "auth-region";
+            public const string SampleDnsSuffix = "dns-suffix";
+        }
+
+        private const string TestRegionSystemName = "us-west-2";
+        private const string TestService = "lambda";
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+            // Reset the RegionEndpoint static maps so that other tests don't use
+            // sample data from this test.
+            RegionEndpoint.Reload(null);
+        }
+
+        [TestMethod]
+        public void Reload()
+        {
+            var initialRegionEndpoint = RegionEndpoint.USWest2;
+            var initialTestServiceEndpoint = initialRegionEndpoint.GetEndpointForService(TestService);
+
+            // Get and Mutate the manifest
+            var rootData = GetEndpointsManifest();
+            ApplySampleValuesToManifest(rootData);
+
+            // Reload with this mutated manifest
+            var json = rootData.ToJson();
+            using (var ms = new MemoryStream(Encoding.UTF8.GetBytes(json)))
+            {
+                RegionEndpoint.Reload(ms);
+            }
+
+            // Verify the SDK resolves regions and endpoints against the mutated data
+            var regionEndpoint = RegionEndpoint.GetBySystemName(TestRegionSystemName);
+            Assert.IsNotNull(regionEndpoint);
+            Assert.AreEqual(SampleValues.SampleRegionDisplayName, regionEndpoint.DisplayName);
+            Assert.AreEqual(SampleValues.SampleDnsSuffix, regionEndpoint.PartitionDnsSuffix);
+            var endpoint = regionEndpoint.GetEndpointForService(TestService);
+            Assert.IsNotNull(endpoint);
+            Assert.AreEqual(SampleValues.SampleAuthRegion, endpoint.AuthRegion);
+            Assert.AreEqual(SampleValues.SampleHostName, endpoint.Hostname);
+
+            // Verify the initial region-endpoint is unaffected
+            Assert.AreNotEqual(initialRegionEndpoint.DisplayName, regionEndpoint.DisplayName);
+            Assert.AreNotEqual(initialTestServiceEndpoint.AuthRegion, endpoint.AuthRegion);
+            Assert.AreNotEqual(initialTestServiceEndpoint.Hostname, endpoint.Hostname);
+        }
+
+        private JsonData GetEndpointsManifest()
+        {
+            using (var stream = GetStream())
+            using (var reader = new StreamReader(stream))
+            {
+                return JsonMapper.ToObject(reader);
+            }
+        }
+
+        private Stream GetStream()
+        {
+            string assemblyLocation = typeof(RegionEndpointProviderV3).Assembly.Location;
+            if (!string.IsNullOrEmpty(assemblyLocation))
+            {
+                string endpointsPath = @"C:\codebase\v3\aws-sdk-net\sdk\src\Core\endpoints.json";
```

Comment:
This path is hardcoded to where you check out the repo. You need to do something using a relative path from this project.

Reply:
How embarrassing - I forgot to circle back to this before wrapping up! I'll get that fixed up.

Reply:
Fixed